taxonomy/models.py

```diff
@@ -50,8 +50,7 @@ class Meta:
     @transaction.atomic
     def save(self, *args, **kwargs):
         """Handle slugs."""
-        if self.id is None or self.name != get_object_or_404(
-                Vocabulary, id=self.id).name:
+        if self.id is None:
```

Comment:
From the code, I'd say that if the vocabulary name is changed, it is desired that the slug be changed.